### PR TITLE
chore(ci): basic pipeline with building, linting and testing

### PR DESCRIPTION
**Pull request by gordonpn**
Run tests for server
Lints both server and client
Builds both server and client

### CODE REVIEW

File: client/src/components/Hero.js [hunk 1]
```diff
@@ -1,4 +1,7 @@
+/* eslint-disable promise/catch-or-return */
+/* eslint-disable promise/always-return */
```
Comment (Mackbellemore) [inline client/src/components/Hero.js:2]: we should probably just fix the rules for these if theyre gonna pop up

Comment (gordonpn) [reply client/src/components/Hero.js:2]: We're straight up gonna get rid of the example Hero when we start developing our frontend

Comment (Mackbellemore) [reply client/src/components/Hero.js:2]: true true  I meant if these are rules you would like to be disabled then do it now

Comment (Mackbellemore) [reply client/src/components/Hero.js:2]: all good though non blocker

**Review by Mackbellemore — APPROVED** [review]

**Review by KarinKazarian — APPROVED** [review]

**Review by SiuYe — APPROVED** [review]
good job @gordonpn

**Pull request merged**